### PR TITLE
Adding future-proof error message matcher for upload hash validation

### PR DESCRIPTION
**Pull request by MewX**

### CODE REVIEW

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java [hunk 1]
```diff
@@ -260,9 +260,13 @@ <T> T makePostRequest(String url, Optional<Map<String, String>> parameters,
    * Note that making this a separate method to avoid polluting throw lists.
    */
   private void maybeRethrowAsUploadError(HttpResponseException e) throws UploadErrorException {
```
Comment (jzacsh) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java:262]: optional nit: would be helpful to pull these strings out to consts and document them, or give them self-documenting names like `GOOG_MISMATCH_API_ERROR_MSG` and `GOOG_MISMATCH_API_ERROR_OLD_MSG`

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java [hunk 2]
```diff
@@ -260,9 +260,13 @@ <T> T makePostRequest(String url, Optional<Map<String, String>> parameters,
...
-    if (e.getStatusCode() == 400 && e.getContent()
-        .contains("Checksum from header does not match received payload content.")) {
-      throw new UploadErrorException(ERROR_HASH_MISMATCH, e);
+    if (e.getStatusCode() == 400) {
```
Comment (jzacsh) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java:263]: optional nit: your choice, but maybe this is clearer as a single expression? `e.getStatusCode() == 400 && (e.getContent().contains(GOOG_...) || e.getContent().contains(GOOG_...))`

I think this makes your comment about delegation a bit clearer (really _everything_ will be delegated to be handled eleswhere - even the non-400s).

**Review by jzacsh — CHANGES_REQUESTED** [review]
could you add test coverage for this new edge case in to your existing tests? `GooglePhotosImporterTest.java`

**Review by jzacsh — CHANGES_REQUESTED** [review]
could you add test coverage for this new edge case in to your existing tests? `GooglePhotosImporterTest.java`

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java (hunk 1 above)
Comment (MewX) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java:264]: Thanks! Updated.

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java (hunk 2 above)
Comment (MewX) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java:265]: Thanks! I feel like grouping those error messages in one line doesn't make much difference. Actually, how about I update the coment to explicitly mention other 400 errors and non-400 errors will be delegated?

**Comment by MewX** [comment]
> could you add test coverage for this new edge case in to your existing tests? `GooglePhotosImporterTest.java`

Currently `GooglePhotosImporterTest` only receives the mocked `UploadErrorException` signal, which has been tested.

This new error string comes from network request, and should be in `GooglePhotosInterfaceTest`. However, it doesn't exist yet. Due to time limit, can I skip creating this test from scratch 😂 Thanks!

**Comment by jzacsh** [comment]
> > could you add test coverage for this new edge case in to your existing tests? `GooglePhotosImporterTest.java`
> 
> Currently `GooglePhotosImporterTest` only receives the mocked `UploadErrorException` signal, which has been tested.
> 
> This new error string comes from network request, and should be in `GooglePhotosInterfaceTest`. However, it doesn't exist yet. Due to time limit, can I skip creating this test from scratch joy Thanks!

ah that's unfortunate - I've filed #1158 - thanks for pointing this out!
